src/qibolab/instruments/zhinst.py

```diff
@@ -1297,7 +1298,41 @@ def sweep_recursion(self, qubits, couplers, exp, exp_calib, exp_options):
             else:
                 self.select_exp(exp, qubits, couplers, exp_options)
 
-    def sweep_recursion_nt(self, qubits, couplers, options, exp, exp_calib):
+    def find_instrument_address(
+        self, quantum_element: Union[Qubit, Coupler], parameter: str
+    ) -> str:
+        """Find path of the instrument connected to a specified line and
+        qubit/coupler.
+
+        Args:
+            quantum_element (Qubit | Coupler): qubits or couplers on which perform the near time sweep.
+            parameter (str): parameter on which perform the near time sweep.
+        """
+        line_names = {
+            "bias": "flux",
+            "amplitude": "drive",
+        }
+        line_name = line_names[parameter]
+        channel_uid = (
+            self.device_setup.logical_signal_groups[f"q{quantum_element.name}"]
+            .logical_signals[f"{line_name}_line"]
+            .physical_channel.uid
+        )
+        channel_name = channel_uid.split("/")[0]
+        instruments = self.device_setup.instruments
+        for instrument in instruments:
+            if instrument.uid == channel_name:
+                return instrument.address
+        raise NotImplementedError  # to be replaced with a different exception
```

Comment:
```suggestion
        raise RuntimeError(f"Could not find instrument for {quantum_element} {line_name}")
```

Is this good enough replacement?